client/src/components/NewBlockForm.tsx

```diff
@@ -5,12 +5,48 @@ import { valueToDatabaseJSON } from "../lib/slateParser";
 import _ = require("lodash");
 import { FormControl } from "react-bootstrap";
 
+import { getExportsFromNodeAndDescendants } from "../slate-helpers/slate-utils/getInlinesFromNodeAndDescendants";
+
 export class NewBlockForm extends React.Component<any, any> {
   public blockEditor;
+  public numOfPointerExports;
 
   public constructor(props: any) {
     super(props);
     this.state = this.blankState();
+    this.numOfPointerExports = 0;
+  }
+
+  public shouldComponentUpdate(nextProps: any, nextState: any) {
+    // initially, this.state.blockValue is an empty object {}
+    // if it hasn't been updated yet, we just update as usual
+    if (typeof nextState.blockValue.toJSON !== "function") {
+      return true;
+    }
+
+    // otherwise we'll check to see if the number of
+    // exported pointers has changed
+
+    const valueAsJSON = nextState.blockValue.toJSON();
+    const pointerExports: any[] = getExportsFromNodeAndDescendants(valueAsJSON.document);
+
+    // if the number of pointers has changed, we send
+    // the pointers up the React hierarchy to the EpisodeShowPage
+    // to add to availablePointers
+    if (this.numOfPointerExports !== pointerExports.length) {
+      this.props.setExportsInNewQuestionForm(pointerExports);
+      this.numOfPointerExports = pointerExports.length;
+
+      // we can return false here because the change to availablePointers
+      // up the hierarchy will trigger a new update
+      return false;
+    }
+
+    return true;
```

Comment:
This seems like a pretty messy solution to me. Generally, data shouldn't be passed up from a child to a parent, like this. It's duplicating state. 

I suggest instead moving the data from the new form into redux, rather than having blockvalue stay in it. This wasn't needed before, but feels like it's needed now.

Reply:
Seperately, I like that you left comments, but would have preferred small functions with nice names :)